client/layout/masterbar/logged-in.jsx

```diff
@@ -201,140 +223,272 @@ class MasterbarLoggedIn extends Component {
 		);
 	}
 
-	render() {
-		const isWordPressActionSearchFeatureEnabled = config.isEnabled( 'wordpress-action-search' );
-		const {
-			domainOnlySite,
-			translate,
-			isCheckout,
-			isCheckoutPending,
-			isMigrationInProgress,
-			previousPath,
-			siteSlug,
-			isJetpackNotAtomic,
-			title,
-			currentSelectedSiteSlug,
-			currentSelectedSiteId,
-		} = this.props;
+	handleToggleMenu = () => {
+		this.setState( { isMenuOpen: ! this.state.isMenuOpen } );
+	};
 
-		const { isActionSearchVisible } = this.state;
+	dismissPopover = () => {
+		this.props.savePreference( MENU_POPOVER_PREFERENCE_KEY, true );
+	};
 
-		if ( isCheckout || isCheckoutPending ) {
-			return (
+	checkout = () => {
```

Comment:
On a side note, is it necessary to use arrow functions for all those rendering methods? Or can they be regular old-school function-like methods?

Reply:
I prefer the arrow way because it automatically binds the function to `this`. But I don't think any of these are event listeners to benefit from that.